packages/aws-cdk/lib/advisory.ts

```diff
@@ -0,0 +1,39 @@
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import { warning } from './logging';
+
+export function printAdvisories() {
+  const advisories: Advisory[] = [
+    new NetmaskAdvisory(),
+  ];
+
+  const messages = advisories.map(a => a.message()).filter(m => m) as string[];
+  if (messages.length > 0) {
+    warning('We have identified the following advisories that may be applicable to your CDK project.');
+    messages.forEach(m => warning(`• ${m}`));
+  }
+}
+
+interface Advisory {
+  message(): string | void;
+}
+
+class NetmaskAdvisory implements Advisory {
+  public message(): string | void {
+    const pkgLockJsonPath = path.join(process.cwd(), 'package-lock.json');
+    const pkgJsonPath = path.join(process.cwd(), 'package.json');
+
+    if (fs.existsSync(pkgLockJsonPath)) {
```

Comment:
Doesn't it make more sense to check for a `yarn.lock` instead to DO do the work, rather than the presence of a `package-lock.json` to NOT do the work?